### ISSUE TITLE
Confusing behaviour when casting with required fields

### ISSUE THREAD

**Issue by jtrees** [hrzndhrn/xema#156]
The cast behaviour when using required properties is a bit confusing.

In the example below I try two different input values on the same schema. They are both successfully casted when the schema is adapted to have no required properties (that part isn't in the example code).

However, when a property is required in atom form, the variant with the string form fails, despite the fact that the cast operation would actually convert the string to an atom anyway.

## Example

```elixir
defmodule Foo do
  defstruct [:msg]
end

defmodule XemaTest do
  import Xema.Builder

  def run(input), do: Xema.cast(schema, input)

  defp schema do
    [
      module: Foo,
      properties: %{msg: :string},
      required: [:msg]
    ]
    |> strux()
    |> Xema.new()
  end
end

XemaTest.run(%{:msg => "hello"})
# => {:ok, %Foo{msg: "hello"}}

XemaTest.run(%{"msg" => "hello"})
# => {:error,
#     %Xema.CastError{
#       error: nil,
#       key: nil,
#       message: nil,
#       path: [],
#       required: [:msg],
#       to: Foo,
#       value: %{"msg" => "hello"}
#     }}
```

**Comment by NickNeck** [comment]
Thanks for reporting. That looks like a bug, and I will take a look in the next few days.

**Comment by NickNeck** [comment]
I have published `0.14.1` that should fix the problem you have. 
One tip, `xema` can create a schema and a struct in one step:
```elixir
iex(5)> defmodule Foo do
...(5)>   use Xema
...(5)>
...(5)>   xema do
...(5)>     field :message, :string
...(5)>     required [:message]
...(5)>   end
...(5)> end
...
iex(6)> Foo.cast!(%{"message" => "bar"})
%Foo{message: "bar"}
```
Have a nice weekend.

**Comment by jtrees** [comment]
Wow, that was fast! Thanks for the fix and thanks for the tip!
